src/lib/dnssd/minimal_mdns/responders/IP.cpp

```diff
@@ -16,18 +16,24 @@
  */
 #include "IP.h"
 
+#include <lib/dnssd/minimal_mdns/AddressPolicy.h>
 #include <lib/dnssd/minimal_mdns/records/IP.h>
 
 namespace mdns {
 namespace Minimal {
 
+using chip::Platform::UniquePtr;
+
 void IPv4Responder::AddAllResponses(const chip::Inet::IPPacketInfo * source, ResponderDelegate * delegate,
                                     const ResponseConfiguration & configuration)
 {
     chip::Inet::IPAddress addr;
-    for (chip::Inet::InterfaceAddressIterator it; it.HasCurrent(); it.Next())
+    UniquePtr<IpAddressIterator> ips = Policies::GetIpAddressesForEndpoint(source->Interface);
+    VerifyOrDie(ips);
+
+    while (ips->Next(addr))
     {
-        if ((it.GetInterfaceId() == source->Interface) && (it.GetAddress(addr) == CHIP_NO_ERROR) && addr.IsIPv4())
+        if (addr.IsIPv4())
```

Comment:
This seems costly to iterate over all addresses twice, rather than having a separate method to interate over IPv4 addresses and IPv6 addresses, and where someone could decide to not implement IPv4? They could still use the same logic under the hood and just filter, if that works well for them, but if it works fine to iterate separately for a platform, that could be accomodated as well.